### ISSUE TITLE
`cd ..` doesn't work well when directory which is deleted

### ISSUE THREAD

**Issue by yangchenyun** [xonsh/xonsh#1107]
```bash
cd /tmp
mkdir -p foo/bar and cd foo/bar
rm -rf /tmp/foo
cd ..
FileNotFoundError: [Errno 2] No such file or directory
```

Possible solution:
in `fish-shell`, when `rm` is executed, it clears the directory stack.
in `bash`, `cd ..` still moves up but raise an error until a valid parent directory is found. 

<img width="783" alt="screen shot 2016-06-04 at 10 31 03 pm" src="https://cloud.githubusercontent.com/assets/1043403/15803788/0d234fb0-2aa4-11e6-8957-0f69bf4cd102.png">

**Comment by adqm** [comment]
Thanks for reporting, @yangchenyun!  It looks like what we are doing is consistent with Bash, but that fish might be doing something nicer (though I'm not sure clearing the directory stack is necessarily the ultimate "right" answer).  I'll investigate...

**Comment by adqm** [comment]
Oh wait, no you're right.  Ours is totally broken.  I kind of like the bash approach.

**Comment by scopatz** [comment]
Thanks again for reporting @yangchenyun! I have noticed something very similar when using version control and removing directories that I am currently in.

**Comment by scopatz** [comment]
@yangchenyun, @adqm has implemented a fix for this, would you mind trying it out.  It has been merged to master

**Comment by adqm** [comment]
I believe this was fixed in #1144, which has been merged into master.  I'm going to close this issue for now.  If this persists, please feel free to re-open!
